wordpress-mobile/WordPress-Android
# P2 people invite links

Part of #13456

This PR introduce possibility for P2 sites to create/share/disable invite links for the roles on their site.


<p align=center><image src="https://user-images.githubusercontent.com/47797566/107771650-59155380-6d3b-11eb-9950-855ec831f29a.gif" width=300/></p>


Hey @mattmiklic wanted to check with you couple of deviations from original, let me know if it's ok (and as always any other feedback welcome 🙇 ):

- As shown in the gif above, introduced loading views while transitioning between states
- As shown below, I used for the Expire the date format we have already in the People screen

| ![image](https://user-images.githubusercontent.com/47797566/107806085-9d6a1900-6d66-11eb-8e94-a20100244bc4.png) | ![image](https://user-images.githubusercontent.com/47797566/107806121-ae1a8f00-6d66-11eb-84cf-ebd59c96bc0b.png) |
|---|---|



Note: a dedicated PR is on the way for analytics and unit testing.

## To test
First of all check the Invite Link section is only available for P2 sites, then use a P2 site for below tests

### Generating links
- On a P2 site with no invitation links already generated tap on Generate new link
- Check the links are created and the roles are all the roles available on the web for that site
- Check the web reports those links now
- Using the app share link button copy the links and check they are consistent with those available on the web for every role

### Sharing an invite link
- Check you can share every role link and copy it or open it in one of the apps available in the share bottom sheet
- Check the shared link is consistent
- Try to use it with another user and see the link allows you to join the site in the expected role (check you are also able to join as administrator)

### Disabling links
- Keep a copy of an invite link
- Disable the links and check you are landed again in the screen with Generate new link button
- Try to use the previously copied link and check it's no more valid

### Snackbars
- Try to make an action for invite links with flight mode on (for example try to generate or delete links)
- Check you get a snackbar to notify the user about the issue

### Smoke test
Smoke test the Invite People screen overall, including the already present people invitation mechanism


PR submission checklist:
